tests/utils.py

```diff
@@ -0,0 +1,123 @@
+from inspect import signature
+
+import torch
+
+
+def model_probably_good(
+        model: torch.nn.Module, shape_in: tuple[int, ...], shape_out: tuple[int, ...]
+) -> bool:
+    """Checks whether the model output has expected shape, is probably unconstrained, and the model
+    has its configs stored.
+
+    This function generates dummy data with a padded batch dimension on top of the
+    input dimension (so ``shape_in`` should exclude a batch dimension). The data is passed through
+    the ``model``. Subsequent tests are described in ``_shapes_match``, ``_probably_unconstrained``,
+    and ``_has_correct_config``.
+
+    Args:
+        model (torch.nn.Module): The module to be tested.
+        shape_in (tuple[int, ...]): Input data shape excluding the batch dimension.
+        shape_out (tuple[int, ...]): Output data shape excluding the batch dimension.
+
+    Returns:
+        bool: Indicator for whether the model meets the three conditions above.
+    """
+    bs = 100
+    x = torch.randn((bs, ) + shape_in)
```

Comment:
Never too fond of randomness in tests, if it can be avoided. 🫤

Reply:
I think hard-coded are more likely to miss bugs than this though. Do u have a suggestion for what hard-coded value would be useful here? (Or alternatives to this test)

Reply:
Or maybe this is not the right way to test these modules...